src/hooks/useSendChatRequest.ts

```diff
@@ -179,10 +179,23 @@ export const useSendChatRequest = () => {
   );
 
   const submit = useCallback(
-    (question: string, maybeMode?: LspChatMode) => {
+    ({
+      question,
+      maybeMode,
+      maybeMessages,
+    }: {
+      question?: string;
+      maybeMode?: LspChatMode;
+      maybeMessages?: ChatMessage[];
+    }) => {
+      if (!question && !maybeMessages) return;
       // const message: ChatMessage = { role: "user", content: question };
-      const message: UserMessage = maybeAddImagesToQuestion(question);
-      const messages = messagesWithSystemPrompt.concat(message);
+      const message: UserMessage = question
+        ? maybeAddImagesToQuestion(question)
+        : ({} as UserMessage);
```

Comment:
should this be cast ? 

I think this can be avoided by having the arguments typed as

```ts
: {question: string, maybeMode?: LspChatMode} 
|  {maybeMode?:LspChatMode, messages: ChatMessage[]} 
```